src/main/java/com/example/report/service/ReportService.java

```diff
@@ -31,53 +42,68 @@ public Map<String, BigDecimal> calculator(BigDecimal amount, Employee employee,
+            calculatedNumbers.put(PERSONAL_INCOME_TAX, amount.multiply(BigDecimal.valueOf(-0.2)));
+            calculatedNumbers.put(GROSS_AMOUNT, amount.multiply(BigDecimal.valueOf(-1)));
             return calculatedNumbers;
-        } else if (!isPensionsPayer && benefitType.equals("Accrual") && calculationMethod.equals("Gross")) {
-            calculatedNumbers.put("netAmount", amount.multiply(BigDecimal.valueOf(0.8)));
-            calculatedNumbers.put("pensionsFund", BigDecimal.valueOf(0));
-            calculatedNumbers.put("personalIncomeTax", amount.multiply(BigDecimal.valueOf(0.2)));
-            calculatedNumbers.put("grossAmount", amount);
+        } else if (!isPensionsPayer && benefitType.equals(ACCRUAL) && calculationMethod.equals(GROSS)) {
+            calculatedNumbers.put(NET_AMOUNT, amount.multiply(BigDecimal.valueOf(0.8)));
+            calculatedNumbers.put(PENSIONS_FUND, BigDecimal.valueOf(0));
+            calculatedNumbers.put(PERSONAL_INCOME_TAX, amount.multiply(BigDecimal.valueOf(0.2)));
+            calculatedNumbers.put(GROSS_AMOUNT, amount);
             return calculatedNumbers;
-        } else if (!isPensionsPayer && benefitType.equals("Accrual") && calculationMethod.equals("Net")) {
-            calculatedNumbers.put("netAmount", amount);
-            calculatedNumbers.put("pensionsFund", BigDecimal.valueOf(0));
-            calculatedNumbers.put("personalIncomeTax", amount.multiply(BigDecimal.valueOf(0.25)));
-            calculatedNumbers.put("grossAmount", amount.multiply(BigDecimal.valueOf(1/0.8)));
+        } else if (!isPensionsPayer && benefitType.equals(ACCRUAL) && calculationMethod.equals(NET)) {
+            calculatedNumbers.put(NET_AMOUNT, amount);
+            calculatedNumbers.put(PENSIONS_FUND, BigDecimal.valueOf(0));
+            calculatedNumbers.put(PERSONAL_INCOME_TAX, amount.multiply(BigDecimal.valueOf(0.25)));
+            calculatedNumbers.put(GROSS_AMOUNT, amount.divide(BigDecimal.valueOf(0.8), RoundingMode.HALF_UP));
             return calculatedNumbers;
         } else {
-            calculatedNumbers.put("netAmount", amount.multiply(BigDecimal.valueOf(-1)));
-            calculatedNumbers.put("pensionsFund", BigDecimal.valueOf(0));
-            calculatedNumbers.put("personalIncomeTax", amount.multiply(BigDecimal.valueOf(0.2)));
-            calculatedNumbers.put("grossAmount", amount.multiply(BigDecimal.valueOf(-1)));
+            calculatedNumbers.put(NET_AMOUNT, amount.multiply(BigDecimal.valueOf(-1)));
+            calculatedNumbers.put(PENSIONS_FUND, BigDecimal.valueOf(0));
+            calculatedNumbers.put(PERSONAL_INCOME_TAX, amount.multiply(BigDecimal.valueOf(0.2)));
+            calculatedNumbers.put(GROSS_AMOUNT, amount.multiply(BigDecimal.valueOf(-1)));
             return calculatedNumbers;
         }
 
     }
 
+    public ReportEntry documentEntryToReportEntry(Document document, Report report){
```

Comment:
This method is most probably not used in other classes, let's make it private if so